tests/DocumentationExamplesTest.php

```diff
@@ -0,0 +1,943 @@
+        }
+        $this->assertInventoryCount(3);
+    }
+
+    public function testExample_6_13()
+    {
+        $db = new Database($this->manager, $this->getDatabaseName());
+
+        // Start Example 6
+        $insertManyResult = $db->inventory->insertMany([
+            [
+                'item' => 'journal', 
+                'qty' => 25,
+                'size' => ['h' => 14, 'w' => 21, 'uom' => 'cm'],
+                'status' => 'A',
+            ],
+            [
+                'item' => 'notebook', 
+                'qty' => 50,
+                'size' => ['h' => 8.5, 'w' => 11, 'uom' => 'in'],
+                'status' => 'A',
+            ],
+            [
+                'item' => 'paper', 
+                'qty' => 100,
+                'size' => ['h' => 8.5, 'w' => 11, 'uom' => 'in'],
+                'status' => 'D',
+            ],
+            [
+                'item' => 'planner', 
+                'qty' => 75,
+                'size' => ['h' => 22.85, 'w' => 30, 'uom' => 'cm'],
+                'status' => 'D',
+            ],
+            [
+                'item' => 'postcard', 
+                'qty' => 45,
+                'size' => ['h' => 10, 'w' => 15.25, 'uom' => 'cm'],
+                'status' => 'A',
+            ],
+        ]);
+        // End Example 6
+
+        $this->assertSame(5, $insertManyResult->getInsertedCount());
+        foreach ($insertManyResult->getInsertedIds() as $id) {
+            $this->assertInstanceOf('MongoDB\BSON\ObjectID', $id);
+        }
+        $this->assertInventoryCount(5);
+
+        // Start Example 7
+        $cursor = $db->inventory->find([]);
+        // End Example 7
+
+        $this->assertCursorCount(5, $cursor);
+
+        // Start Example 8
+        $cursor = $db->inventory->find();
+        // End Example 8
+
+        $this->assertCursorCount(5, $cursor);
+
+        // Start Example 9
+        $cursor = $db->inventory->find(['status' => 'D']);
+        // End Example 9
+
+        $this->assertCursorCount(2, $cursor);
+
+        // Start Example 10
+        $cursor = $db->inventory->find(['status' => ['$in' => ['A', 'D']]]);
+        // End Example 10
+
+        $this->assertCursorCount(5, $cursor);
+
+        // Start Example 11
+        $cursor = $db->inventory->find([
+            'status' => 'A',
+            'qty' => ['$lt' => 30],
+        ]);
+        // End Example 11
+
+        $this->assertCursorCount(1, $cursor);
+
+        // Start Example 12
+        $cursor = $db->inventory->find([
+            '$or' => [
+                ['status' => 'A'],
+                ['qty' => ['$lt' => 30]],
+            ],
+        ]);
+        // End Example 12
+
+        $this->assertCursorCount(3, $cursor);
+
+        // Start Example 13
+        $cursor = $db->inventory->find([
+            'status' => 'A',
+            '$or' => [
+                ['qty' => ['$lt' => 30]],
+                // Alternatively: ['item' => new \MongoDB\BSON\Regex('^p')]
```

Comment:
@kay-kim: Is this kosher? I noticed Perl and Python were using `$regex` instead of their BSON regex class.